### PR TITLE
arch/sim: Fix hostfs.h No such file or directory in makedep

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>

## Summary
regression by commit:
https://github.com/apache/incubator-nuttx/commit/7e5b0f81e93c7e879ce8434d57e8bf4e2319c1c0

### CODE REVIEW

File: arch/sim/src/Makefile [hunk 1]
```diff
@@ -38,6 +38,7 @@ include $(TOPDIR)/Make.defs
 
 ARCH_SRCDIR = $(TOPDIR)$(DELIM)arch$(DELIM)$(CONFIG_ARCH)$(DELIM)src
 
+INCLUDES += ${shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)}
```
Comment (patacongo) [inline arch/sim/src/Makefile:41]: $(ARCH_SRCDIR) would be $(TOPDIR)/arch/sim/src.  Why do you expect hostft.h to be there?

Comment (patacongo) [reply arch/sim/src/Makefile:41]: hostfs.h is located at $(TOPDIR)/include/nuttx/hostfs.h

Or is there something else going on that is not clear?

Comment (patacongo) [reply arch/sim/src/Makefile:41]: Okay, yes, I see in arch/sim/src/Makefile:

    213 hostfs.h: $(TOPDIR)/include/nuttx/fs/hostfs.h
    214         @echo "CP:  $<"
    215         $(Q) cp $< $@

Comment (xiaoxiang781216) [reply arch/sim/src/Makefile:41]: Yes, up_hostfs.c is built in host environment and then hostfs.h is copied locally for finding easlier.

**Pull request merged**